### PR TITLE
Move upload translations action to own  workflow

### PR DESCRIPTION
**Pull request by McGiverGim**
Our current workflow that uploads the translations to Crowdin is executed during the nightly build. This has a problem: if we merge several PRs at the same time, we start several nightly builds without any control of what of them finish first. 
We can upload some updated messages file to Crowdin, and after that, upload some old messages files from an earlier merge, that ends deleting the new strings until other merge with the correct strings end.

This PR fixes that:
- Moves the upload translation to it's own execution, out of the nightly
- Filters by file modified, to execute only when there are changes in the `locales/en/` files
- Uses concurrency to have only one execution at a time, giving priority to the latest merge

### CODE REVIEW

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> FAIL
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> FAIL
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

**Review by haslinghuis — APPROVED** [review]

**Review by blckmn — APPROVED** [review]

**Pull request merged**
